File: kotlin/1413_minimum_value_to_get_positive_step_by_step_sum.kt
// Runtime: 144 ms
// Memory Usage: 33.5 MB

class Solution {
    fun minStartValue(nums: IntArray): Int {
        var res = 1
        var sum = 1

        for(num in nums) {
            sum += num
            if(sum < 1) {
                res += 1 - sum
                sum = 1
            }
        }

        return res
    }
}

File: php/1413_minimum_value_to_get_positive_step_by_step_sum.php
<?php
// Runtime: 4 ms
// Memory Usage: 15.7 MB

class Solution {
    /**
     * @param int[] $nums
     * @return int
     */
    function minStartValue($nums) {
        $curr = 1;
        $res = 1;
        foreach ($nums as $num) {
            $curr += $num;
            if ($curr < 1) {
                $res += 1 - $curr;
                $curr = 1;
            }
        }
        
        return $res;
    }
}